### PR TITLE
Fix spell formulas, effect & client debug issues in 7.72

### PR DESCRIPTION
**Pull request by dnvd**
Fix spell formulas, effect & client debug issues in 7.72
This has been up for discussion if it should be fixed or not #45.

BUT...
The PR also includes a few crash fixes since some of the types threw an error when using some of the runes/spell.
Force wave, energy beams, HMMs, LMMs etc now also uses the correct effect for this version.

### CODE REVIEW

File: data/spells/scripts/attack/poison_storm.lua [hunk 1]
```diff
@@ -4,10 +4,26 @@ combat:setParameter(COMBAT_PARAM_EFFECT, CONST_ME_GREEN_RINGS)
 
 combat:setArea(createCombatArea(AREA_CIRCLE6X6))
 
+local condition = createConditionObject(CONDITION_POISON)
+setConditionParam(condition, CONDITION_PARAM_DELAYED, 1)
```
Comment (nekiro) [inline data/spells/scripts/attack/poison_storm.lua:8]: please use tfs 1.x equivalents. This is old pre 1.x coding style.
setConditionParam is for example condition:setParameter
addDamageCondition is condition:addDamage

**Review by nekiro — CHANGES_REQUESTED** [review]

**Comment by dnvd** [comment]
Ok, made some changes.

File: data/spells/scripts/attack/berserk.lua [hunk 2]
```diff
@@ -6,9 +6,9 @@ combat:setParameter(COMBAT_PARAM_USECHARGES, true)
 combat:setArea(createCombatArea(AREA_SQUARE1X1))
 
 function onGetFormulaValues(player, skill, attack, factor)
-	local min = (player:getLevel() / 5) + (skill * attack * 0.03) + 7
-	local max = (player:getLevel() / 5) + (skill * attack * 0.05) + 11
-	return -min, -max
+	min = -(level * 2.2)
```
Comment (nekiro) [inline data/spells/scripts/attack/berserk.lua:9]: this value is global for some reason
same with max
use local before the names

File: data/spells/scripts/attack/poison_storm.lua [hunk 3]
```diff
@@ -4,10 +4,26 @@ combat:setParameter(COMBAT_PARAM_EFFECT, CONST_ME_GREEN_RINGS)
...
+condition:addDamage(2, 4000, -30)
+condition:addDamage(3, 5000, -20)
+condition:addDamage(3, 5000, -10)
+condition:addDamage(3, 5000, -7)
+condition:addDamage(3, 5000, -5)
+condition:addDamage(4, 5000, -4)
+condition:addDamage(6, 5000, -3)
+condition:addDamage(9, 5000, -2)
+condition:addDamage(12, 5000, -1)
+addCombatCondition(combat, condition)
```
Comment (nekiro) [inline data/spells/scripts/attack/poison_storm.lua:21]: combat:addCondition(condition)

File: data/spells/scripts/attack/poison_storm.lua [hunk 4]
```diff
@@ -4,10 +4,26 @@ combat:setParameter(COMBAT_PARAM_EFFECT, CONST_ME_GREEN_RINGS)
 
 combat:setArea(createCombatArea(AREA_CIRCLE6X6))
 
+local condition = createConditionObject(CONDITION_POISON)
```
Comment (nekiro) [inline data/spells/scripts/attack/poison_storm.lua:7]: Condition(CONDITION_POISON)

File: data/spells/scripts/attack/poison_storm.lua [hunk 5]
```diff
@@ -4,10 +4,26 @@ combat:setParameter(COMBAT_PARAM_EFFECT, CONST_ME_GREEN_RINGS)
...
+condition:addDamage(6, 5000, -3)
+condition:addDamage(9, 5000, -2)
+condition:addDamage(12, 5000, -1)
+addCombatCondition(combat, condition)
+
 function onGetFormulaValues(player, level, magicLevel)
-	local min = (level / 5) + (magicLevel * 4) + 75
-	local max = (level / 5) + (magicLevel * 10) + 150
-	return -min, -max
+	min = -((level * 2) + (magicLevel * 3)) * 0.9
```
Comment (nekiro) [inline data/spells/scripts/attack/poison_storm.lua:24]: same as above, restore local, these two are globals now

**Review by nekiro — CHANGES_REQUESTED** [review]

**Pull request closed without merging**